sources/System/Collision/CollisionSystem.cpp

```diff
@@ -0,0 +1,47 @@
+//
+// Created by Louis Auzuret on 5/20/21
+//
+
+#include "Component/Position/PositionComponent.hpp"
+#include "Component/Collision/CollisionComponent.hpp"
+#include "System/Collision/CollisionSystem.hpp"
+
+namespace BBM
+{
+	CollisionSystem::CollisionSystem(WAL::Wal &wal)
+	: WAL::System({typeid(PositionComponent), typeid(CollisionComponent)}), _wal(wal)
+	{
+
+	}
+	void CollisionSystem::onFixedUpdate(WAL::Entity &entity)
+	{
+		try {
+			auto &posA = entity.getComponent<PositionComponent>();
+			auto &col = entity.getComponent<CollisionComponent>();
+			Vector3f minA = {	std::min(posA.getX(), posA.getX() + col.getBoundX()),
+								std::min(posA.getY(), posA.getY() + col.getBoundY()),
+								std::min(posA.getZ(), posA.getZ() + col.getBoundZ())};	
+			Vector3f maxA = {	std::max(posA.getX(), posA.getX() + col.getBoundX()),
+								std::max(posA.getY(), posA.getY() + col.getBoundY()),
+								std::max(posA.getZ(), posA.getZ() + col.getBoundZ())};
+			for (auto &other : _wal.scene->getEntities()) {
+				if (&other == &entity)
+					continue;
+				auto &colB = entity.getComponent<CollisionComponent>();
```

Comment:
If you encounter an entity that does not have a CollisionComponent, NotFoundException will be raised.